llvm/test/CodeGen/AMDGPU/idot4s.ll

```diff
@@ -3450,4 +3450,850 @@ entry:
 }
 
 
+define amdgpu_kernel void @ByteOffsetCorrectness(ptr addrspace(1) noalias readonly align 16 %inptr0, ptr addrspace(1) noalias readonly align 16 %inptr1, ptr addrspace(1) noalias align 16 %inptr2) local_unnamed_addr {
```

Comment:
Comment the test. Can this be shrunk?

Reply:
The behavior is actually a bit difficult to reproduce with a small concise example -- optimizations / DAG building consistently transform the code in a way which hides the issue.

Reply:
But what transformations? You can usually massage the test to defeat them. e.g. use volatile loads.

If you avoid using the work item ID intrinsics, you can avoid more transforms